include/swift/Runtime/Metadata.h

```diff
@@ -2643,23 +2649,47 @@ swift_getFunctionTypeMetadata(const void *flagsArgsAndResult[]);
 SWIFT_RUNTIME_EXPORT
 const FunctionTypeMetadata *
 swift_getFunctionTypeMetadata1(FunctionTypeFlags flags,
-                               const void *arg0,
+                               const Metadata *arg0,
                                const Metadata *resultMetadata);
 
 SWIFT_RUNTIME_EXPORT
 const FunctionTypeMetadata *
+swift_getFunctionTypeMetadata1WithFlags(FunctionTypeFlags flags,
+                                        const Metadata *arg0,
+                                        const uint32_t *paramFlags,
+                                        const Metadata *resultMetadata);
+
+SWIFT_RUNTIME_EXPORT
+const FunctionTypeMetadata *
 swift_getFunctionTypeMetadata2(FunctionTypeFlags flags,
-                               const void *arg0,
-                               const void *arg1,
+                               const Metadata *arg0,
+                               const Metadata *arg1,
                                const Metadata *resultMetadata);
 
 SWIFT_RUNTIME_EXPORT
 const FunctionTypeMetadata *
-swift_getFunctionTypeMetadata3(FunctionTypeFlags flags,
-                               const void *arg0,
-                               const void *arg1,
-                               const void *arg2,
-                               const Metadata *resultMetadata);
+swift_getFunctionTypeMetadata2WithFlags(FunctionTypeFlags flags,
+                                        const Metadata *arg0,
+                                        const Metadata *arg1,
+                                        const uint32_t *paramFlags,
```

Comment:
This should take two uint32_t's.

Reply:
If we are going to go this route I going to re-arrange it so it’s param1, flags1, param2, flags2 etc.

Reply:
Yes, that seems reasonable.